x-pack/plugins/transform/public/app/hooks/use_delete_transform.tsx

```diff
@@ -4,52 +4,278 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import React from 'react';
-
+import React, { useCallback, useEffect, useState } from 'react';
 import { i18n } from '@kbn/i18n';
+import { CoreSetup } from 'kibana/public';
 import { toMountPoint } from '../../../../../../src/plugins/kibana_react/public';
-
-import { TransformEndpointRequest, TransformEndpointResult } from '../../../common';
-
-import { getErrorMessage } from '../../shared_imports';
-
+import {
+  TransformEndpointRequest,
+  DeleteTransformEndpointResult,
+  DeleteTransformStatus,
+} from '../../../common';
+import { getErrorMessage, extractErrorMessage } from '../../shared_imports';
 import { useAppDependencies, useToastNotifications } from '../app_dependencies';
 import { TransformListRow, refreshTransformList$, REFRESH_TRANSFORM_LIST_STATE } from '../common';
 import { ToastNotificationText } from '../components';
-
 import { useApi } from './use_api';
+import { API_BASE_PATH } from '../../../common/constants';
+import { IIndexPattern } from '../../../../../../src/plugins/data/common/index_patterns';
+
+export const canDeleteIndex = async (http: CoreSetup['http']) => {
+  const privilege = await http.get(`${API_BASE_PATH}privileges`);
+  if (!privilege) {
+    return false;
+  }
+  return privilege.hasAllPrivileges;
+};
```

Comment:
It would be great to have `IndexService` which contains `canDeleteIndex ` method.

Reply:
`checkIndexPatternExists` could be presented in that service as well. It will allow to extract the logic from the react hook.